contracts/contracts/FleekERC721.sol

```diff
@@ -527,6 +533,7 @@ contract FleekERC721 is Initializable, ERC721Upgradeable, FleekAccessControl {
      *
      * - the tokenId must be minted and valid.
      * - the sender must have the `tokenOwner` role.
+     * - the contract must be not paused.
      *
      */
     function burn(uint256 tokenId) public virtual requireTokenRole(tokenId, Roles.Owner) {
```

Comment:
the `whenNotPaused` modifier is forgotten here.
Also, we need to discuss when we will use this feature and decide on what functions we want to work on when the contract is paused.

Reply:
It is on `_beforeTokenTransfer` function that is called for mint, transfer and burn functions.